test/tests/performance/__init__.py

```diff
@@ -35,11 +35,73 @@
 # "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
 # trademark licensing policies.
 
+import math
+from math import sqrt
 from ptl.utils.pbs_testsuite import *
 
 
 class TestPerformance(PBSTestSuite):
     """
     Base test suite for Performance tests
     """
-    pass
+
+    def mean(self, lst):
+        """calculates mean"""
+        return sum(lst) / len(lst)
+
+    def stddev(self, lst):
+        """returns the standard deviation of lst"""
+        mn = self.mean(lst)
+        variance = sum([(e - mn) ** 2 for e in lst])
+        return sqrt(variance)
+
+    def check_value(self, res):
+        if isinstance(res, list):
+            for val in res:
+                if isinstance(val, list):
+                    self.check_value(val)
+                    continue
+                if not isinstance(val, (int, float)):
+                    raise self.failureException(
+                        "Test result list must be int or float")
+        else:
+            if not isinstance(res, (int, float)):
+                raise self.failureException("Test result must be int or float")
+
+    def perf_test_result(self, result, test_measure, unit):
+        """
+        Add test results to json file. If a multiple trial values are passed
+        calculate mean,std_dev,min,max for the list.
+        """
+        self.check_value(result)
+        if isinstance(result, list) and len(result) > 1:
+            mean_res = self.mean(result)
+            mean_res = round(mean_res, 2)
+            stddev_res = self.stddev(result)
+            stddev_res = round(stddev_res, 2)
+            max_res = round(max(result), 2)
+            min_res = round(min(result), 2)
+            trail_no = 1
+            trial_data = []
+            for trial_result in result:
+                trial_result = round(trial_result, 2)
+                trial_data.append(
+                    {"trial_no": trail_no, "value": trial_result})
+                trail_no += 1
+            test_data = {"test_measure": test_measure,
```

Comment:
Two fields 'test_measure' and 'unit' should not be under 'test_data' dictionary. All the other fields 'mean', 'std_dev', 'minimum', 'maximum' & 'trials' should be under 'test_data' field as separate sub dictionary. This has been followed correctly in _else_ case but not here.